.circleci/config.yml

```diff
@@ -41,48 +75,58 @@ jobs:
                 event: fail
                 template: basic_fail_1
                 branch_pattern: main,release/.+
-      - save_cache:
-          key: ct-modcache-v2-{{ checksum "go.mod" }}
-          paths:
-            - /go/pkg/mod
 
-  vault_integration_tests:
+  vault-integration-tests:
     executor:
       name: go
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - ct-modcache-v2-{{ checksum "go.mod" }}
-      - run:
-          name: Install Vault
-          command: |
-            curl -sLo vault.zip https://releases.hashicorp.com/vault/${VAULT_VERSION}/vault_${VAULT_VERSION}_linux_amd64.zip
-            unzip vault.zip
-            sudo cp vault /usr/local/bin/
-            vault version
+      - setup-vault
       - run: |
           go test -count=1 -timeout=80s -tags 'integration vault' ./... -run Vault
-      - save_cache:
-          key: ct-modcache-v2-{{ checksum "go.mod" }}
-          paths:
-            - /go/pkg/mod
 
-  e2e_tests:
+  unit-tests:
```

Comment:
TODO: Remove

Reply:
Addressed in https://github.com/hashicorp/consul-terraform-sync/pull/869/commits/b6e8e0ee61a8137207278705c7c473ebb52d612e